craft_store/publisher/_publishergw.py

```diff
@@ -79,6 +94,57 @@ def _check_error(response: httpx.Response) -> None:
             brief, store_errors=errors.StoreErrorList(error_list)
         )
 
+    def list_registered_names(
+        self, include_collaborations: bool = False
+    ) -> list[models.RegisteredNameModel]:
+        """Return names registered by the authenticated user.
+
+        :param include_collaborations: if True, includes names the user is a
+            collaborator on but does not own.
+        :returns: A sequence of names registered to the user.
+
+        API docs: https://api.charmhub.io/docs/default.html#list_registered_names
+        """
+        response = self._client.get(
+            f"/v1/{self._namespace}",
+            params={"include-collaborations": include_collaborations},
+        )
+        self._check_error(response, expected_keys={"results"})
+        return [
+            models.RegisteredNameModel.unmarshal(item)
+            for item in response.json()["results"]
+        ]
```

Comment:
Probably out-of-scope, so this'll just be a comment:

Any time `self._check_error()` doesn't raise an error, an immediate next step is to call `response.json()` in the caller to use the now-validated response. However, `self._check_error()` calls `response.json()` already to check for errors. Could it be refactored to return this already-parsed JSON and save some time?